### ISSUE TITLE
No way to get maximum value of stats from definitions only?

### ISSUE THREAD

**Issue by bhollis** [Bungie-net/api#448]
If I look up items in the manifest definitions, and look at their stats, there are properties `value`, `maximum`, and `minimum`. However, it appears that `maximum` is always equal to `value`. Furthermore, the stat definition itself has no `maximum`. I think I'm misinterpreting what `DestinyInventoryItemStatDefinition.maximum` means - it seems like it's the maximum value that stat can be *for that particular item*. If that's the case, I'd like to request a `maximum` property on `DestinyStatDefinition` that spells out the maximum possible value for the stat (so we can draw it as a bar).

**Comment by floatingatoll** [comment]
I wanted to take a try at restating this as a set of questions using different terminology, since I think these are all useful to consider here:

Do the item’s minimum/maximum stats reflect only the range of stat values that can be influenced by standard perk choices on the item?

Do masterwork items have different minimum and maximum values from their non-masterwork equivalents?

Do armor mods affect the minimum and maximum values of any items?

**Comment by bhollis** [comment]
Those are all good questions. But there's another, underlying question that I'm looking to answer here - for a stat like "Recovery", what is the maximum *possible* value for any item? What's the ceiling?

**Comment by floatingatoll** [comment]
In your view, should armor items list their intrinsic unmodifiable maximum *separately* from the perk bubbles and from the three possible sets of + modifiers for light, medium, and heavy armor? Or should they list the maximum *given* all three – intrinsic, selectable perks, and armor type?

I’m inclined to ask why the maximum stat is included at all, since 3-way combinatorics is a rather tall order for a value labeled “maximum”.

**Comment by bhollis** [comment]
Can you make a piece of armor have a stat higher than 5? I think the global maximum is 5. So I'd like the stat definition to tell me the maximum is 5.

**Comment by vthornheart-bng** [comment]
Yeah, so ultimately this is a holdover from the original D1 Armory.  IIRC, the idea was that minimum was the lowest level we saw an item spawn at, value was the "base" stats at which the item spawned, and "maximum" was the item with talent nodes activated such that the stat was maximized for the rolled talent nodes.

Currently, maximum is fairly meaningless: I don't think I have altered it yet to actually account for sockets rather than talent nodes - but it's been such a low priority that I've not had the time to make the changes necessary for such a switch-over.  If I were to do it, I'd probably set it to the same standard that we had in D1: where maximum would represent the stat if you maximized it via selections of reusable plugs (I wouldn't account for non-reusable plugs, as that could theoretically represent a much larger world of options and isn't relevant to the intrinsic properties of the item itself in isolation).

I don't think I'd want to rely on the maximum possible value that you can boost an item to in theory, i.e. the perceived global maximum.  That's subject to change on a whim, and could theoretically be modified by factors that are extrinsic to the item itself.  But I would like to change maximum to at least mean what it can reach with its own, intrinsic properties.  I don't see it getting onto our docket anytime soon however.  I will document the Maximum field appropriately, letting people know it is deprecated and should not be used until/unless we fix how it is calculated.
